app/validation/validator.py

```diff
@@ -204,6 +207,38 @@ def validate_repeat_rule_restricted(rule, block):
 
         return errors
 
+
+    def validate_calculated_summary_type(self, block, answers_with_parent_ids):
+        answer_types = [
+            answers_with_parent_ids[answer_id]['answer']['type']
+            for answer_id in block['calculation']['answers_to_calculate']
+        ]
+
+        if not all(type == answer_types[0] for type in answer_types):
```

Comment:
Would be best not to override the builtin `type` variable here and line 227, 236

Reply:
Yes, stupid of me, will call it answer_type.